hera_cal/abscal.py

```diff
@@ -1075,66 +1202,74 @@ def abscal_run(data_files, model_files, calfits_infiles=None, verbose=True, over
+
+    # make blank gains if no modelfiles
+    else:
+        gain_pols = set(flatten(map(lambda p: [p[0], p[1]], data_pols)))
+        gain_dict = map(lambda p: map(lambda a: ((a,p), np.ones((Ntimes, Nfreqs), np.complex)), data_ants), gain_pols)
+        gain_dict = odict(flatten(gain_dict))
+        flag_dict = odict(map(lambda k: (k, np.ones((Ntimes, Nfreqs), np.bool)), gain_dict.keys()))
+
+    # write to file
+    if write_calfits:
+        # load calfits file if provided
+        if calfits_infile is not None:
+            cal_in = UVCal()
+            cal_in.read_calfits(calfits_infile)
+            out_gains, out_flags = io.load_cal(cal_in)
+            for k in out_gains.keys():
+                if k in gain_dict:
+                    out_gains[k] *= gain_dict[k]
+                    out_flags[k] += flag_dict[k]
+                else:
+                    out_flags[k] += np.ones(out_flags[k], np.bool)
+            io.update_cal(cal_in, output_calfits_path,  gains=out_gains, flags=out_flags, add_to_history=history, clobber=overwrite)
+        # write a calfits from scratch
         else:
-            gain_pols = set(flatten(map(lambda p: [p[0], p[1]], data_pols)))
-            gain_dict = map(lambda p: map(lambda a: ((a,p), np.ones((Ntimes, Nfreqs), np.complex)), data_ants), gain_pols)
-            gain_dict = odict(flatten(gain_dict))
-            flag_dict = odict(map(lambda k: (k, np.ones((Ntimes, Nfreqs), np.bool)), gain_dict.keys()))
-
-        # write to file
-        if write_calfits:
-            # load calfits file if provided
-            if calfits_infiles is not None:
-                cal_in = UVCal()
-                cal_in.read_calfits(calfits_infiles[i])
-                out_gains, out_flags = io.load_cal(cal_in)
-                for k in out_gains.keys():
-                    if k in gain_dict:
-                        out_gains[k] *= gain_dict[k]
-                        out_flags[k] += flag_dict[k]
-                    else:
-                        out_flags[k] += np.ones(out_flags[k], np.bool)
-                io.update_cal(cal_in, output_calfits_fname,  gains=out_gains, flags=out_flags, add_to_history=history, clobber=overwrite)
-            # write a calfits from scratch
-            else:
-                io.write_cal(output_calfits_fname, gain_dict, data_freqs, data_times, flags=flag_dict, quality=None, 
-                     total_qual=None, return_uvc=False, overwrite=overwrite, history=history)
-
-        # append gain dict to gains
-        gains.append(gain_dict)
+            io.write_cal(output_calfits_path, gain_dict, data_freqs, data_times, flags=flag_dict, quality=None, 
+                 total_qual=None, return_uvc=False, overwrite=overwrite, history=history)
 
     # form return tuple
     return_obj = ()
 
     # return gains if desired
     if return_gains:
-        return_obj += (gains,)
+        return_obj += (gain_dict,)
```

Comment:
why was this changed?

Reply:
`gains` is no longer defined. It used to hold a list of `gain_dict`s, one per data file. But now there's just one data file.

Reply:
ah, oh yeah that's right.. okay cool